test/integration_tests/test_uploads.py

```diff
@@ -122,6 +122,6 @@ def test_label_upload(with_group_and_file_data, api_as_admin):
 
     data.files['metadata'] = metadata
     r = api_as_admin.post('/upload/label', files=data.files)
-    assert r.status_code == 400
+    assert r.status_code == 500
```

Comment:
to fix this properly catch the right error on line 79 of `upload.py`. I am guessing it is not an AttributeError, most likely a TypeError.